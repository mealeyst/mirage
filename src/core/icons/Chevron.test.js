import React from 'react'
import { css } from 'styled-components'
import 'jest-styled-components'

import { Chevron } from 'SRC'
import { animated, pulseDirection, validateDirection } from './Chevron'

const { mountWithTheme } = global

describe('(Styled Component) Chevron', () => {
  const createChevron = (props) => {
    return mountWithTheme(<Chevron {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createChevron({down: true}))
    .toMatchSnapshot()

    expect(createChevron({up: true}))
    .toMatchSnapshot()

    expect(createChevron({left: true}))
    .toMatchSnapshot()

    expect(createChevron({right: true}))
    .toMatchSnapshot()

    expect(createChevron({down: true, animated: true}))
    .toMatchSnapshot()

    expect(createChevron({up: true, animated: true}))
    .toMatchSnapshot()

    expect(createChevron({left: true, animated: true}))
    .toMatchSnapshot()

    expect(createChevron({right: true, animated: true}))
    .toMatchSnapshot()
  })

  test('animating the chevron', () => {
    expect(createChevron({down: true, animated: true}))
    .toHaveStyleRule({
      modifer: css`${animated}`
    })
  })

  describe('(Custome Style Function) pulseDirection', () => {
    test('setting the direction to the right', () => {
      expect(pulseDirection({right: true}))
      .toEqual('.2rem')
    })
    test('setting the direction to the left', () => {
      expect(pulseDirection({left: true}))
      .toEqual('-.2rem')
    })
  })

  describe('(Custom Prop Validation) validateDirection', () => {
    test('setting no directions should return an error', () => {
      expect(validateDirection({}, '', 'Chevron'))
      .toEqual(Error('No direction prop was supplied to Chevron, please select one.'))
    })

    test('setting more than one direction should return an error', () => {
      expect(validateDirection({up: true, down: true}, '', 'Chevron'))
      .toEqual(Error('More than one direction prop was supplied to Chevron, only use one.'))
    })

    test('setting only one direction should return null', () => {
      expect(validateDirection({up: true}, '', 'Chevron'))
      .toBeNull()
    })
  })
})
