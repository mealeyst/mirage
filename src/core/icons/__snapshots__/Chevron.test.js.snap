// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Styled Component) Chevron matching the snapshot 1`] = `
.c0 {
  width: 16px;
  height: 8px;
  fill: none;
  stroke: #00003C;
  stroke-miterlimit: 10;
  stroke-width: 4px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  cursor: pointer;
  margin: 10px;
}

.c0:hover polyline {
  stroke-width: 7px;
}

<Styled(BaseChevron)
  down={true}
  ignoreErrors={false}
>
  <BaseChevron
    className="c0"
    down={true}
    ignoreErrors={false}
  >
    <svg
      className="c0"
      viewBox="0 0 48 24.12"
    >
      <polyline
        points="1 1 24 22.75 47 1"
      />
    </svg>
  </BaseChevron>
</Styled(BaseChevron)>
`;

exports[`(Styled Component) Chevron matching the snapshot 2`] = `
.c0 {
  width: 16px;
  height: 8px;
  fill: none;
  stroke: #00003C;
  stroke-miterlimit: 10;
  stroke-width: 4px;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  cursor: pointer;
  margin: 10px;
}

.c0:hover polyline {
  stroke-width: 7px;
}

<Styled(BaseChevron)
  ignoreErrors={false}
  up={true}
>
  <BaseChevron
    className="c0"
    ignoreErrors={false}
    up={true}
  >
    <svg
      className="c0"
      viewBox="0 0 48 24.12"
    >
      <polyline
        points="1 1 24 22.75 47 1"
      />
    </svg>
  </BaseChevron>
</Styled(BaseChevron)>
`;

exports[`(Styled Component) Chevron matching the snapshot 3`] = `
.c0 {
  width: 16px;
  height: 8px;
  fill: none;
  stroke: #00003C;
  stroke-miterlimit: 10;
  stroke-width: 4px;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  cursor: pointer;
  margin: 10px;
}

.c0:hover polyline {
  stroke-width: 7px;
}

<Styled(BaseChevron)
  ignoreErrors={false}
  left={true}
>
  <BaseChevron
    className="c0"
    ignoreErrors={false}
    left={true}
  >
    <svg
      className="c0"
      viewBox="0 0 48 24.12"
    >
      <polyline
        points="1 1 24 22.75 47 1"
      />
    </svg>
  </BaseChevron>
</Styled(BaseChevron)>
`;

exports[`(Styled Component) Chevron matching the snapshot 4`] = `
.c0 {
  width: 16px;
  height: 8px;
  fill: none;
  stroke: #00003C;
  stroke-miterlimit: 10;
  stroke-width: 4px;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  cursor: pointer;
  margin: 10px;
}

.c0:hover polyline {
  stroke-width: 7px;
}

<Styled(BaseChevron)
  ignoreErrors={false}
  right={true}
>
  <BaseChevron
    className="c0"
    ignoreErrors={false}
    right={true}
  >
    <svg
      className="c0"
      viewBox="0 0 48 24.12"
    >
      <polyline
        points="1 1 24 22.75 47 1"
      />
    </svg>
  </BaseChevron>
</Styled(BaseChevron)>
`;

exports[`(Styled Component) Chevron matching the snapshot 5`] = `
.c0 {
  width: 16px;
  height: 8px;
  fill: none;
  stroke: #00003C;
  stroke-miterlimit: 10;
  stroke-width: 4px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  cursor: pointer;
  -webkit-animation: gtBSkN 0.75s infinite;
  animation: gtBSkN 0.75s infinite;
  margin: 10px;
}

.c0:hover polyline {
  stroke-width: 7px;
}

<Styled(BaseChevron)
  animated={true}
  down={true}
  ignoreErrors={false}
>
  <BaseChevron
    animated={true}
    className="c0"
    down={true}
    ignoreErrors={false}
  >
    <svg
      className="c0"
      viewBox="0 0 48 24.12"
    >
      <polyline
        points="1 1 24 22.75 47 1"
      />
    </svg>
  </BaseChevron>
</Styled(BaseChevron)>
`;

exports[`(Styled Component) Chevron matching the snapshot 6`] = `
.c0 {
  width: 16px;
  height: 8px;
  fill: none;
  stroke: #00003C;
  stroke-miterlimit: 10;
  stroke-width: 4px;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  cursor: pointer;
  -webkit-animation: ktSMsp 0.75s infinite;
  animation: ktSMsp 0.75s infinite;
  margin: 10px;
}

.c0:hover polyline {
  stroke-width: 7px;
}

<Styled(BaseChevron)
  animated={true}
  ignoreErrors={false}
  up={true}
>
  <BaseChevron
    animated={true}
    className="c0"
    ignoreErrors={false}
    up={true}
  >
    <svg
      className="c0"
      viewBox="0 0 48 24.12"
    >
      <polyline
        points="1 1 24 22.75 47 1"
      />
    </svg>
  </BaseChevron>
</Styled(BaseChevron)>
`;

exports[`(Styled Component) Chevron matching the snapshot 7`] = `
.c0 {
  width: 16px;
  height: 8px;
  fill: none;
  stroke: #00003C;
  stroke-miterlimit: 10;
  stroke-width: 4px;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  cursor: pointer;
  -webkit-animation: hGBtOi 0.75s infinite;
  animation: hGBtOi 0.75s infinite;
  margin: 10px;
}

.c0:hover polyline {
  stroke-width: 7px;
}

<Styled(BaseChevron)
  animated={true}
  ignoreErrors={false}
  left={true}
>
  <BaseChevron
    animated={true}
    className="c0"
    ignoreErrors={false}
    left={true}
  >
    <svg
      className="c0"
      viewBox="0 0 48 24.12"
    >
      <polyline
        points="1 1 24 22.75 47 1"
      />
    </svg>
  </BaseChevron>
</Styled(BaseChevron)>
`;

exports[`(Styled Component) Chevron matching the snapshot 8`] = `
.c0 {
  width: 16px;
  height: 8px;
  fill: none;
  stroke: #00003C;
  stroke-miterlimit: 10;
  stroke-width: 4px;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  cursor: pointer;
  -webkit-animation: lovGSO 0.75s infinite;
  animation: lovGSO 0.75s infinite;
  margin: 10px;
}

.c0:hover polyline {
  stroke-width: 7px;
}

<Styled(BaseChevron)
  animated={true}
  ignoreErrors={false}
  right={true}
>
  <BaseChevron
    animated={true}
    className="c0"
    ignoreErrors={false}
    right={true}
  >
    <svg
      className="c0"
      viewBox="0 0 48 24.12"
    >
      <polyline
        points="1 1 24 22.75 47 1"
      />
    </svg>
  </BaseChevron>
</Styled(BaseChevron)>
`;
